app/controllers/concerns/metadata.rb

```diff
@@ -17,20 +17,21 @@ def metadata(opts = {})
       options = {}
       options[:meta] = { total: opts[:total], search: params[:search],
                          limit: pagination_limit, offset: pagination_offset }
-      options[:links] = links(opts[:total])
+      options[:links] = links(last_offset(opts[:total]))
       options
     end
 
-    def links(count)
+    def links(last_offset)
       base_url = "#{ENV['PATH_PREFIX']}/#{ENV['APP_NAME']}"\
         "/#{controller_name}"
-      last_offset = (count / pagination_limit.to_f).ceil
       {
         first: "#{base_url}?limit=#{pagination_limit}&offset=1",
         last: "#{base_url}?limit=#{pagination_limit}&offset=#{last_offset}",
-        previous: previous_link(base_url),
-        next: next_link(base_url, last_offset)
-      }
+        previous: (previous_link(base_url) if pagination_offset > 1),
```

Comment:
This is nefarious of me, but the previous link should not ever link to an invalid/empty page.

```
$ http --verify no https://ci.foo.redhat.com:1337/api/compliance/rules limit==2 offset==518
{
    "data": [],
    "links": {
        "first": "/api/compliance/rules?limit=2&offset=1",
        "last": "/api/compliance/rules?limit=2&offset=514",
        "previous": "/api/compliance/rules?limit=2&offset=517"
    },
    "meta": {
        "limit": 2,
        "offset": 518,
        "search": null,
        "total": 1027
    }
}
```

Reply:
:smile: